src/main/java/com/avispl/symphony/dal/infrastructure/management/tripleplay/TriplePlayAggregatorCommunicator.java

```diff
@@ -0,0 +1,601 @@
+				String responseOfQueryRequest = doPost(TriplePlayURL.BASE_URI, requestBodyV2.buildRequestBody());
+				MonitoringData monitoringData = objectMapper.readValue(responseOfQueryRequest, MonitoringData.class);
+				Client responseClient = monitoringData.getClientWrapper().getClients().get(0);
+
+				if (value != responseClient.getActivity().getCurrentService().getName()) {
+					throw new IllegalStateException(String.format("Can not control channel of device %s, unknow error", client.getDeviceId()));
+				}
+				Map<String, String> properties = client.getProperties();
+				List<AdvancedControllableProperty> advancedControllableProperties = client.getControllableProperties();
+
+				Client cachedClient = cachedClients.get(getKeyInAggregatedDevice(client));
+				List<String> serviceNames = new ArrayList<>();
+
+				//update service in cachedClients
+				for (AdvancedControllableProperty advancedControllableProperty : client.getControllableProperties()) {
+					if (advancedControllableProperty.getName() == buildNameForChannel(ChannelInfoMetric.SELECT_CHANNEL.getName())) {
+						client.getProperties().put(buildNameForChannel(ChannelInfoMetric.LAST_CHANNEL.getName()), advancedControllableProperty.getValue().toString());
+						for (Service service : cachedClient.getServices()) {
+							serviceNames.add(service.getName());
+							if (service.getName() == client.getProperties().get(buildNameForChannel(ChannelInfoMetric.LAST_CHANNEL.getName()))) {
+								cachedClient.getActivity().setLastService(service);
+							}
+							if (service.getName() == value) {
+								cachedClient.getActivity().setCurrentService(service);
+							}
+						}
+						break;
+					}
+				}
+
+				addAdvanceControlProperties(advancedControllableProperties, properties,
+						ControllablePropertyFactory.createDropdown(buildNameForChannel(ChannelInfoMetric.SELECT_CHANNEL.getName()),
+								serviceNames, value));
+
+				client.setProperties(properties);
+				client.setControllableProperties(advancedControllableProperties);
+
+				cachedAggregatedDevices.put(client.getDeviceId(), client);
+			} else {
+				throw new IllegalStateException(String.format("Can not change channel of device %s", client.getDeviceId()));
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * This method is used for calling control channel properties
+	 *
+	 * @param property name of controllable property
+	 * @param value value of controllable property
+	 * @param client device need control
+	 */
+	private void channelControl(String property, String value, AggregatedDevice client) {
+		ChannelInfoMetric channelInfoMetric = ChannelInfoMetric.getByName(property);
+		switch (channelInfoMetric) {
+			case SELECT_CHANNEL:
+				selectChannelControl(value, client);
+		}
+	}
+
+	/**
+	 * Because amino device doesn't have serivces(Maybe amino die is an issue), so I use cachedServices instead.
```

Comment:
Because amino device doesn't have serivces(Maybe amino die is an issue), so I use cachedServices instead. It should be 
=> providing the list of services from cachedServices

Reply:
I fixed it